src/main.rs

```diff
@@ -1,6 +1,91 @@
-use std::error::Error;
+use std::{error::Error, fs::File, io::{Read, self, BufReader}, mem::size_of, u8, usize, os::unix::prelude::{AsFd, FromRawFd, AsRawFd}};
+
+use clap::{Parser, command};
+
+mod blktrace {
+    #![allow(non_upper_case_globals)]
+    #![allow(non_camel_case_types)]
+    #![allow(non_fmt_panics)]
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+	pub fn blk_tc_act(act: u32) -> u32 {
+		act << BLK_TC_SHIFT
+	}
+}
+
+static STDIN_PATH: &str = "-";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+	/// Input file path, - for stdin
+	#[arg(short, long, default_value_t = STDIN_PATH.to_string(),)]
+	input: String,
+}
+
+impl Args {
+	pub fn is_stdin(&self) -> bool {
+		self.input.eq(&STDIN_PATH)
+	}
+}
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
+    const FRAGMENT_SIZE: usize = size_of::<blktrace::blk_io_trace>();
```

Comment:
OK. I wonder if there is a less hardcoded way to do all this. Something to discuss for V0.2.